specs/DesignDiscussions/Template.md

```diff
@@ -0,0 +1,418 @@
+# Overview
+## Templating
+![](assets/Template/img1.png)
+**Figure 1**
+
+Template allows users to define where data binding occurs.
+
+```json
+{
+    "type": "TextBlock",
+    "text": "{latestPrice} ",
+    "size": "ExtraLarge"
+}
+```
+
+Once binding is defined, adaptive cards will be updated by changing only the data.
+```json
+{
+  "latestPrice": 128.9,
+}
+```
+### Syntax
+- Binding syntax starts with { and ends with }. E.g., {myProperty}
+- Dot-notation to access sub-objects
+ - {employee.manager.name}
+- Indexer syntax to retrieve properties by key or items in an array
+ - {employee.peers[0].name}
+- $data assigns data context to current element
+- $when conditional expression that determines if current element to be included.
+- functions are provided to help SDK users
+
+# Parsing
+We use ANTLR to generate parser and lexer
+
+## ANTLR
+From a grammar, ANTLR generates a parser that can build parse trees and also generates a listener interface (or visitor) that makes it easy to respond to the recognition of phrases of interest.
+ANTLR is really two things: a tool that translates your grammar to a parser/lexer in Java (or other target language) and the runtime needed by the generated parsers/lexers. .NET and JS version will handle server side tasks. We will use C++ as target languange and shared across Android, iOS, and UWP as we do currently with our shared json parser.
+Once parsers and lexers are generated, the tool is no longer needed, so it won't be part of template SDK. AEL uses ANTLR, so AEL SDK already includes ANTRL runtime. There will be no additional penaly of using ANTLR by ensuring that we use the same version of the ANTLR runtime
```

Comment:
>AEL [](start = 106, length = 3)

What is AEL? You are referencing an Acronym and a component that you havent defined prior in an Index section nor have a hyperlink here for somebody without the AEL context to come in and understand the reference